src/Symfony/Component/DependencyInjection/Compiler/RepeatedPass.php

```diff
@@ -56,13 +56,14 @@ public function __construct(array $passes)
      */
     public function process(ContainerBuilder $container)
     {
+repeat:
         $this->repeat = false;
         foreach ($this->passes as $pass) {
             $pass->process($container);
         }
 
         if ($this->repeat) {
-            $this->process($container);
+            goto repeat;
```

Comment:
please add a comment saying that goto is used to optimize performance here. Otherwise, we will get repeated PRs doing the opposite change